tests/topotests/bgp_distance_change/test_bgp_admin_dist.py

```diff
@@ -0,0 +1,1285 @@
+    configured same as ebgp learnt route
+
+TC_2:
+    Verify ebgp admin distance functionality with ECMP.
+
+TC_3:
+    Verify ibgp admin distance functionality when static route is
+    configured same as bgp learnt route.
+TC_4:
+    Verify ibgp admin distance functionality with ECMP.
+
+TC_7: Chaos - Verify bgp admin distance functionality with chaos.
+"""
+
+#################################
+# TOPOLOGY
+#################################
+"""
+
+                    +-------+
+         +--------- |  R2   |
+         |          +-------+
+         |iBGP           |
+     +-------+           |
+     |  R1   |           |iBGP
+     +-------+           |
+         |               |
+         |    iBGP   +-------+   eBGP   +-------+
+         +---------- |  R3   |----------|  R4   |
+                     +-------+          +-------+
+                        |
+                        |eBGP
+                        |
+                    +-------+
+                    |  R5   |
+                    +-------+
+
+
+"""
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+
+# Required to instantiate the topology builder class.
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    step,
+    write_test_footer,
+    create_static_routes,
+    verify_rib,
+    create_route_maps,
+    create_prefix_lists,
+    check_address_types,
+    reset_config_on_routers,
+    check_router_status,
+    stop_router,
+    kill_router_daemons,
+    start_router_daemons,
+    start_router,
+    get_frr_ipv6_linklocal,
+    verify_fib_routes,
+    required_linux_kernel_version
+)
+from lib.topolog import logger
+from lib.bgp import (
+    verify_bgp_convergence,
+    create_router_bgp,
+    verify_best_path_as_per_admin_distance,
+    clear_bgp,
+)
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+from lib.topojson import build_config_from_json
+from lib.topolog import logger
+
+# Global variables
+topo = None
+bgp_convergence = False
+pytestmark = [pytest.mark.bgpd, pytest.mark.staticd]
+
+NETWORK = {
+    "ipv4": [
+        "50.0.20.1/32",
+        "50.0.20.2/32",
+        "150.0.20.1/32",
+        "150.0.20.2/32",
+        "10.0.20.1/32",
+        "10.0.20.2/32",
+    ],
+    "ipv6": [
+        "50::1/128",
+        "50::2/128",
+        "150::1/128",
+        "150::2/128",
+        "1::1/128",
+        "1::2/128",
+    ],
+}
+
+ADDR_TYPES = check_address_types()
+
+
+def setup_module(mod):
+    """
+    Sets up the pytest environment
+
+    * `mod`: module name
+    """
+
+    # Required linux kernel version for this suite to run.
+    result = required_linux_kernel_version("4.15")
```

Comment:
What's so specific here to skip for older kernels?

Reply:
I just removed this check. These kernel versions were added to other suits, i picked them up, but as pointed they are not applicable to this test. So deleted them.